File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(a, b):
    n = min(a, b)
    while a % n != 0 or b % n != 0:
        n -= 1
    return n


a,b = map(int, input().split())
d = gcd(a, b)
print(d)
print(a * b // d)

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
const fs = require('fs');

const [a,b] = fs.readFileSync('dev/stdin').toString().trim().split(" ").map(Number);
const aa = Math.min(a,b);
const bb = Math.max(a,b);
const gcd = () => {
    for (let i = aa; i >= 1; i--){
        if (aa % i === 0 && bb % i === 0) {
            return i;
        }
    }
}
const lcm = () => {
    let temp1 = aa
    let temp2 = bb
    
    while(temp1 != temp2){
        if (temp1 < temp2){
            temp1 += aa
        } else{
            temp2 += bb
        }
    }
    
    return temp1;
}

console.log(gcd())
console.log(lcm())